### PR TITLE
Added manage-users endpoint with Intergration Tests

### PR DESCRIPTION
**Pull request by badekk**
## Added manage-users endpoint with Intergration Tests

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

- [ ] Unit test(s)
- [X] Integration test(s)
- [ ] Tested manually

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings
- [] I have added frontend code and checked it for mobile & desktop

### CODE REVIEW

**Comment by mar-pan** [comment]
Task linked: [CU-862j9cr7n List](https://app.clickup.com/t/862j9cr7n)

File: src/backend/Auth/Actions/AuthActions/ManageUsersActions/ManageUsersResponseItem.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Auth.Actions.AuthActions.ManageUsersActions
+{
+    public class ManageUsersResponseItem
+    {
+        public string Id { get; set; } = default!; // User Id
+        public string FullName { get; set; } = default!;
+        public string? WarehouseName { get; set; } // can be null
```
Comment (h4570) [inline src/backend/Auth/Actions/AuthActions/ManageUsersActions/ManageUsersResponseItem.cs:13]: Te moje komentarze wywalaj, bo one tylko czysto informacyjnie dla cb były

File: src/backend/AuthTests/Actions/AuthActionsTest/ManageUsersActionsTest/ManageUsersActionTest.cs [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+            var action = toolkit.Resolve<ManageUsersAction>();
+
+            // Arrange
+            var roleDonorId = context.Roles.First(c => c.Name == RoleName.Donor).Id;
+            var roleAdminId = context.Roles.First(c => c.Name == RoleName.Administrator).Id;
+            var roleDelivererId = context.Roles.First(c => c.Name == RoleName.Deliverer).Id;
+
+            var testingUser1 = new User()
+            {
+                Status = DbEntityStatus.Active,
```
Comment (h4570) [inline src/backend/AuthTests/Actions/AuthActionsTest/ManageUsersActionsTest/ManageUsersActionTest.cs:171]: No zrób tą fixture, bo mega śmietnik jest i se narobiłeś roboty, jakbyś zrobił fixture ja początku to bys szybciej to zrobił

**Review by h4570 — CHANGES_REQUESTED** [review]

File: src/backend/Auth/Actions/AuthActions/ManageUsersActions/ManageUsersResponseItem.cs (hunk 1 above)
Comment (badekk) [reply src/backend/Auth/Actions/AuthActions/ManageUsersActions/ManageUsersResponseItem.cs:13]: fixed

File: src/backend/AuthTests/Actions/AuthActionsTest/ManageUsersActionsTest/ManageUsersActionTest.cs (hunk 2 above)
Comment (badekk) [reply src/backend/AuthTests/Actions/AuthActionsTest/ManageUsersActionsTest/ManageUsersActionTest.cs]: fixtured

**Pull request merged**